### ISSUE TITLE
mediaSession artwork not displaying

### ISSUE THREAD

**Issue by m7kra** [martpie/museeks#650]
Dear martpie,

I hope you don't take it wrong. This question is not about Museeks, but about a music player of my own, which I'm trying to build. Feel free to ignore this issue :)

So here is the question, as I described in [Stack Overflow](https://stackoverflow.com/questions/73448782/media-sessions-artwork-not-showing-up): Inspired by your code, I'm trying to access the album cover with `fs`, as `base64` and display it in `mediaSession`'s artwork. This is my code:

```js
const cover = this.getCover(this.playback.album.coverPath);

// Setup media session metadata
let metadata = {
    title: this.playback.track.title,
    album: this.playback.album.title,
    artist: this.playback.album.artist,
    artwork: cover? [{ src: cover }] : null
};
navigator.mediaSession.metadata = new MediaMetadata(metadata);


function getCover(coverPath) {
    const data = fs.readFileSync(coverPath, { encoding: 'base64' });
    const format = path.extname(coverPath).substr(1);
    return `data:${format};base64,${data}`;
}
```

This is not working on Ubuntu, only on Windows. I know that you code is almost the same, and it works in my OS. Have you gone through the same problem? Have you found any solutions?

Thank you so much,
m7kra

**Comment by m7kra** [comment]
Perhaps I haven't explained myself good enough. I get the cover on a previous step, which I managed to implement. Therefore, I'm sure that `album.coverPath` has the correct path for an album cover. After that, I do the following:

```js
const cover = this.getCover(this.playback.album.coverPath);
```

Perhaps the confusion lays in this step: for me, `this.getCover` is the same as your `getFileAsBase64`, and I implemented it in a similar fashion - no very far from copy/pasting :)

I then set the artwork in the exact same way as you do:

```js
let metadata = {
    title: this.playback.track.title,
    album: this.playback.album.title,
    artist: this.playback.album.artist,
    artwork: cover? [{ src: cover }] : null
};
navigator.mediaSession.metadata = new MediaMetadata(metadata);
```

It is because this approach is so similar to yours that I thought I could ask you if you went through the same problem of the cover not appearing on Ubuntu.

On a personal note, let me express my profound admiration and respect for you attitude. Having someone helping his 'opponent' is very altruistic, and helps to strengthen my feelings for the open-source world, to which I would like to, one day, be a useful contributor. Thank you!

**Comment by martpie** [comment]
So, to be extremely honest, I have switched from elementary OS to macOS something like 5 years ago, and it's been very tricky for me to test on Linux devices (Parallels was quite cool though, but a bit expensive).

I think I have tested this method on Ubuntu, Elementary OS and KDE, but I don't remember facing any issue, all those things were working out of the box.

What was tricky was to get the base64 correct (there's always one thing not working properly, a small comma missing, or something similar. 

My suggestion would be you clone Museeks and put some `console.log`, so you can compare if you the cover src is the same between your music player and Museeks. 🤔 From there you can probably identify what's going wrong.

--- 

Regarding open-source and stuff 😄: all good all fine, I'm a self-taught and owe a lot to the open-source community and the web in general, and I particularly enjoy sharing knowledge, insights, problems, etc. As long as no one gets mad when I get a little bit less active (hitting the bottom of the wave at the very moment haha, so much work at work).

**Comment by martpie** [comment]
Will close, feel free to ping if you have other questions :)

**Comment by m7kra** [comment]
First of all, sorry for taking so long to answer. I was away for some days, got lost in other projects - constraint-based Sudoku solving took me some time :), etc. You closing this issue really was a reminder that I had things to do...

Anyway, today I cloned Museeks and was astonished to see that `mediaSession` wasn't working! I then installed Museeks 0.13.1, not working either. Going back to Museeks 0.11.0, I finally had the covers showing up - this was likely the last version of Museeks I had installed.

So now my story is a little bit different, and the issue is about your app as well: `mediaSession` doesn't work correctly on Linux (at least Ubuntu). What I will try to do is to check for the user's OS and, if it is a Linux distribution, use pure MPRIS. I was also thinking of opening an issue on Electron's GitHub, but realized that not even Google Chrome has `mediaSession`'s artwork working properly.

This is probably a dead end. I just wanted to, again, thank you, and say that the joy I felt because you actually answered me was enough for this trouble to be worth it :)

**Comment by martpie** [comment]
mmmh that's weird, mediaSession should cover mpris usecases (I tested on vanilla ubuntu and elementary), and I deliberatly moved away from `mpris-service` to avoid bundling native dependencies (it's complicated to make all of this work).

I'll have to try on fresh ubuntu VMs, but I suspect an issue with your DE, more than with Museeks. There's actually an easy way to test that: open spotify or youtube music, and see if it displays on mpris. If it does not, then it's a DE issue. If it does, then it's an issue with Museeks.

**Comment by m7kra** [comment]
My Ubuntu installation is nearly new (1 week), because I got messed up with the 22.04 update :| and reinstalled everything to have a clean machine. Actually, that's a proof that the issue doesn't depend on the DE: when I opened this issue, I was using Ubuntu 20.04. Today, that installation has been erased and I'm on Ubuntu 22.04. Furthermore, playing a video on Youtube doesn't show an artwork, but in Windows it does. I believe that it's not an issue with Museeks itself, but with the Chromium Electron uses.
